---
title: "SMS_4947_20_CVA24v_receptor"
author: "Nima Rafati"
date: "5/1/2020"
output: 
    html_document:
      toc : true
---

```{r functions, echo = F, eval = T}
extractRegion<-function(data,chr,start,end,distance)
{
	x<-data
	x.chr<-x[which(x[,1]==chr),]
	x.chr.start<-x.chr[which(x.chr[,2]>=start-distance),]
	x.chr.start.end<-x.chr.start[which(x.chr.start[,2]<=end+distance),]
	return(x.chr.start.end)
}

```

#Description

I have deleted desmoglein 2 (DSG2) protein in HAP1 cells. When we did a single cell sorting we obtained 2 clones of DSG2 deleted HAP1 cells that we call HAP1-F9 and HAP1-H11. I see an effect of the virus infection in one of the clones, in HAP1-F9 but NOT in HAP1-H11. The effect is that virus life cycle is inhibited, thus no progeny virus are produced in HAP1-F9. Both HAP1-F9 and HAP-H11 cells lack DSG2 according to FACS analysis. Basically we want to see what off target effect there is in HAP1 F9 compared to HAP1 WT.
Since we do not have any experience in bioinformatics of whole genome sequencing we want both the sequencing and bioinformatics on what gene(s) in HAP1-F9 that is different from HAP1 WT. WGS will be done by scilifelabs, and we need help analyzing the data from WGS.  
#Data preparation
Downloading dbSNP file from NCBI ```https://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh37p13/VCF/00-All.vcf.gz```.  
```{bash}
  
```

#Processing available calls
We first explroed the vairant calls generated by NGI. This consist of raw small variants (SNP/INDEL). Thus, we filtered the variants by following criteria for SNP and INDEL separately. The filtering plots are also generated. Then the variants passing the criteria are merged and saved.    
```{bash}
#INDEL
gatk VariantFiltration -R /proj/snic2020-16-70/private/SMS_4947_20_CVA24v_receptor/data/meta_data/reference/human_g1k_v37.fasta --variant Raw_Call-INDEL.vcf -O Raw_Call-INDEL-FILTER.vcf -filter "QUAL < 100.0 " --filter-name "QUAL100" -filter "QD < 10.0" --filter-name "QD2" -filter "FS > 10.0" --filter-name "FS10" -filter "ReadPosRankSum < -5.0" --filter-name "ReadPosRankSum-5" -filter "MQRankSum  < -8.0 " --filter-name "MQRankSum-8" -filter "BaseQRankSum < -5.0" --filter-name "BaseQRankSum-5" -filter "SOR > 4.0" --filter-name "SOR-4" 
--genotype-filter-name "DPFilter" --genotype-filter-expression  "DP < 10.0 || DP >300.0" &

#SNP
gatk VariantFiltration -R /proj/snic2020-16-70/private/SMS_4947_20_CVA24v_receptor/data/meta_data/reference/human_g1k_v37.fasta --variant Raw_Call-SNP.vcf -O Raw_Call-SNP-FILTER.vcf -filter "QUAL < 100.0 " --filter-name "QUAL100" -filter "QD < 20.0" --filter-name "QD20" -filter "FS > 10.0" --filter-name "FS10" -filter "MQ < 40.0 " --filter-name "MQ40" -filter "MQRankSum  < -8.0 " --filter-name "MQRankSum-8" -filter "ReadPosRankSum < -5.0" --filter-name "ReadPosRankSum-5" -filter "BaseQRankSum < -10.0" --filter-name "BaseQRankSum-10" -filter "SOR > 3.0" --filter-name "SOR-3" --genotype-filter-name "DPFilter" --genotype-filter-expression  "DP < 10.0 || DP >400.0" &
wait 

module load vcftools
bgzip Raw_Call-INDEL-FILTER.vcf & 
bgzip Raw_Call-SNP-FILTER.vcf &
wait
tabix -p vcf Raw_Call-INDEL-FILTER.vcf.gz &
tabix -p vcf Raw_Call-SNP-FILTER.vcf.gz &
wait

#Merge vcf files
java -jar /sw/apps/bioinfo/picard/2.20.4/rackham/picard.jar SortVcf I=Raw_Call-SNP-FILTER.vcf I=Raw_Call-INDEL-FILTER.vcf O=Raw_Call-FILTER.sort.vcf

#Extract only PASS variants
fgrep '#' Raw_Call-FILTER.sort.vcf >Final_Call-PASS.vcf
awk '($7=="PASS")' Raw_Call-FILTER.sort.vcf >>Final_Call-PASS.vcf 


#INDEL
 "QUAL < 100.0 " 
 "QD < 10.0" 
 "FS > 10.0" 
 "ReadPosRankSum < -5.0" 
 "MQRankSum  < -8.0 " 
 "BaseQRankSum < -5.0" 
 "SOR > 4.0" 
 "DP < 10.0 || DP >300.0"
#SNP
 "QUAL < 100.0 " 
 "QD < 20.0" 
 "FS > 10.0" 
 "MQ < 40.0 " 
 "MQRankSum  < -8.0 " 
 "ReadPosRankSum < -5.0" 
 "BaseQRankSum < -10.0" 
 "SOR > 3.0" 
 "DP < 10.0 || DP >400.0"
```

#New variant calling from scratch
In genotype calls WT sample did not have any calls showing 0/0 which is unexpected. Thus, we run the GATK best practice pipeline. Here we look into reference allele freq:  
```{bash, eval = F, echo = T}
 python   /home/nimar/private/PERL/vcf-parsing/Calculate_Freq_from_VCF_V1.py  -v Final_Call-PASS_snpEff_fixed_het_positions.vcf >Final_Call-PASS_snpEff_fixed_het_positions_freq.bed
```



# New calls  
## Final plot for depth and new calls mutation on DSG2  
```{r, eval = F, echo = T}
library(zoo)
library(ggbio)
library(Homo.sapiens)
library(reshape2)

coordinate <- c(18, 29078027, 29128814)

##Depth 
###From bam files directly
# library(GenomicAlignments)
DSG2_depth <- read.table('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/depth/F9_H11_WT_Q20_DSG2_18_29.06_29.13.depth', header = F)
colnames(DSG2_depth) <- c('Chr','Pos','F9', 'H11', 'WT')
plot(DSG2_depth$Pos, DSG2_depth$F9, type = 'l', bty = 'n', xlab = 'Chr18', ylab = 'raw depth')
rect(xleft = 29078027, xright = 29128814, ybottom = -2, ytop = max(DSG2_depth[,3:6])+10), rgb(0,0,1,alpha=0.1))

Pos <- rep(1:nrow(DSG2_depth),1)
DSG2_depth_1kb <- cbind(Pos,DSG2_depth[,-c(1:2)])
DSG2_depth_1kb <- as.data.frame(rollapply(DSG2_depth_1kb,500,mean,by=500))
DSG2_depth_1kb$Pos <- seq(DSG2_depth$Pos[1], DSG2_depth$Pos[nrow(DSG2_depth)],500)[-1]

###Normalise the depth
genome_avg_depth <- data.frame(Sample = c('F9', 'H11', 'WT'), Avg_depth = c(35.97, 48.49, 34.75))
DSG2_depth_1kb_normalised <- DSG2_depth_1kb
DSG2_depth_1kb_normalised[,2:4] <- DSG2_depth_1kb[,2:4]/(genome_avg_depth$Avg_depth/max(genome_avg_depth$Avg_depth))
region <- melt(DSG2_depth_1kb_normalised[(DSG2_depth_1kb_normalised$Pos >= as.numeric(coordinate[2])),], id.vars = 'Pos')
colnames(region) <- c('Position', 'Sample', 'Depth')


All_sample_depth_normalised <- ggplot(data = region, aes(x = Position, y = Depth, color = Sample)) + geom_line() + ylim(c(0,80)) + theme_classic() + ylab('Normalised Depth(500 bp)') + annotate(geom="text", x=29100000, y=16, label="Deletion(F9)", col = 'red')

##

##Allele frequency
path <- '~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/Variant_Calling/Genotyping/'
Allele_freq <- read.table(paste0(path,'Final_Call-PASS_snpEff_chr8_DSG2_50k_freq.bed'), head = F)
colnames(Allele_freq) <- c('Chr', 'Start', 'End', 'Info', 'Ref', 'Alt', 'QUAL', 'Filter', 'F9', 'H11', 'WT')
region <- extractRegion(data = Allele_freq, chr = coordinate[1], start = as.numeric(coordinate[2]), end = as.numeric(coordinate[3]), distance = 0)
region <- melt(Allele_freq[,c('End', 'F9', 'H11', 'WT')], id.vars = 'End', variable.name = 'Sample')
colnames(region) <- c('Position', 'Sample', 'Ref_Freq')
p_freq <- ggplot(data = region, aes ( x = Position, y = Ref_Freq, color = Sample)) + geom_line() + theme_classic() + ylab('Ref. Freq.')



##Gene model
#load gene symbol : GRanges, one gene/row
data(genesymbol, package = "biovizBase")
#retrieve information of the gene of interest
wh <- genesymbol[c("DSG2")]
wh <- range(wh, ignore.strand = F)
#Plot the different transcripts  for our gene of interest
p.txdb <- autoplot(Homo.sapiens, which = wh)
##Inversion
INV <- data.frame(Sam = c('F9', 'H11', 'H11','WT'), Start = c(29078027, 29078027, 29099847, 29078027), End = c(29128814, 29128814, 29098216, 29128814), Sample = c(1,2,2,3), SV = c('WT', 'WT', 'Inversion', 'WT')) 

SV <- ggplot(INV) + geom_segment(data = INV, aes(x=Start, xend=End, y=Sample, yend=Sample, col = SV), size = 2,
               arrow = arrow(length = unit(0.2, "cm"))) + scale_y_continuous(breaks = c(1,2,3), labels=c('1' = 'F9', '2' = 'H11', '3' = 'WT') ) +  theme(panel.grid = element_blank(), axis.title.y = element_blank(), axis.ticks.y = element_blank(), legend.position = 'bottom')
##
##Plot
gr18 <- GRanges("chr18", IRanges(29078027, 29128814))
pdf('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/results/Summarised_DSG2.pdf', width = 15, height = 15)
tracks(Gene = p.txdb, Allele_Frequency = p_freq, Depth = All_sample_depth_normalised, Structural_variation = SV, heights = c(1,2,2,2)) + theme_bw()
dev.off()
##
```
## All small variants  
Here we compare the allele frequency between samples. Considering the response to infection among samples we used following criteria to select variants with potential differentiation between F9 vs H11/WT:  
- F9 Ref freq <=0.2 while H11 & WT has ref freq >=0.8  
- F9 Ref freq >=0.8 while H11 & WT has ref freq <=0.2  
```{r, eval = F, echo = T}
library(reshape2)
library(cowplot)
#*****Please note that we focus on sites that all samples have calls.  
##Genome wide distribution of ALL variants
path <- '~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/Variant_Calling/Genotyping/'
genome_wide_AF <- read.table(paste0(path,'Final_Call-PASS_snpEff_freq.bed'), header = F)
colnames(genome_wide_AF) <- c('Chr', 'Start', 'End', 'Info', 'Ref', 'Alt', 'QUAL', 'Filter', 'F9', 'H11', 'WT')
###Remove NAs
genome_wide_AF <- genome_wide_AF[complete.cases(genome_wide_AF),]
genome_wide_AF_melt <- melt(genome_wide_AF[,9:11])
colnames(genome_wide_AF_melt) <- c('Sample', 'Ref_Freq')
genome_wide_AF_melt <- genome_wide_AF_melt[!is.na(genome_wide_AF_melt$Ref_Freq),] # remove NA
inset_plot <- ggplot(genome_wide_AF_melt, aes(Ref_Freq, fill = Sample)) + geom_density( alpha = 0.2) + xlab('Reference Allele Frequency') + ylab('Frequency')
main_plot <- ggplot(genome_wide_AF_melt, aes(Ref_Freq, fill = Sample)) + geom_density( alpha = 0.2)+ xlim(0,0.25) + xlab('Reference Allele Frequency') + ylab('Frequency')
plot.with.inset <-
  ggdraw() +
  draw_plot(main_plot) +
  draw_plot(inset_plot, x = 0.2, y = .2, width = .7, height = .7)
pdf(paste0(path, 'Final_Call-PASS_snpEff_freq_distribution.pdf'), width = 10, height = 10)
plot.with.inset
dev.off()
##
##Selecting differnetiated variants
differentiated_sites <- rbind.data.frame(genome_wide_AF[(genome_wide_AF$F9  <= 0.2 &
                                        genome_wide_AF$H11 >= 0.8 &
                                        genome_wide_AF$WT  >= 0.8),],
                                        genome_wide_AF[(genome_wide_AF$F9  >= 0.8 &
                                        genome_wide_AF$H11 <= 0.2 &
                                        genome_wide_AF$WT  <= 0.2),])
differentiated_sites$ID <- paste0(differentiated_sites$Chr, '_', differentiated_sites$End)
differentiated_sites_chr_table <- as.data.frame(table(differentiated_sites$Chr))
colnames(differentiated_sites_chr_table) <- c('Chr', 'Count')
differentiated_sites_chr_table <- differentiated_sites_chr_table[(as.numeric(differentiated_sites_chr_table$Chr) <= 22),]#Selecting only autosomal chromosome
###Ordering by chormosome number
chrOrder <- 1:22
differentiated_sites_chr_table$Chr <- factor(differentiated_sites_chr_table$Chr, levels=chrOrder)
differentiated_sites_chr_table <- differentiated_sites_chr_table[order(differentiated_sites_chr_table$Chr),]
pdf(paste0(path, 'Final_Call-PASS_snpEff_differentiated_variants_chr_distribution.pdf'), width = 10, height = 7)
ggplot(differentiated_sites_chr_table, aes(x=reorder(Chr, 1:22), y=Count)) +  geom_bar(stat="identity") + theme_bw() + xlab('Chromosome') + ggtitle('Frequency of differentiated variants between F9 and H11/WT') + theme(plot.title = element_text(hjust = 0.5))
dev.off()
```

# Functional variation analysis  
In  this part we look at the variants affecting functionality of gene by SNPEff. SNPEff output was converted to table:  
```java -jar  /sw/bioinfo/snpEff/4.3t/rackham/SnpSift.jar extractFields -s ":" -e "." Final_Call-PASS_snpEff.vcf CHROM POS ID REF QUAL GEN[*].GT GEN[*].AD  "ANN[*].EFFECT" "EFF[*].GENE" >Final_Call-PASS_snpEff.ann.table```

Now we parse to generate general statistics:  
```{r eval = F, echo = T}
library(data.table)
library(kable)
path <- '~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/Variant_Calling/Genotyping/'
##Load summary output of snpEff
snpEff_summary <- read.csv(paste0(path,'Final_Call-PASS_snpEff_count_effects.txt'), sep = '\t', header =T)
kable(snpEff_summary)
##
##Frequency of mutations
snpEff_table <- fread(paste0(path, 'Final_Call-PASS_snpEff.ann.table'), header = T) 
colnames(snpEff_table) <- c('Chr', 'Position', 'ID', 'REF', 'QUAL', 'GT', 'AD', 'Effect', 'Gene')
snpEff_table$ID <- paste0(snpEff_table$Chr, '_', snpEff_table$Position)

### All mutations
effects <- unlist(strsplit(snpEff_table$Effect, ':'))
genes <- unlist(strsplit(snpEff_table$Gene, ':'))
and_index <- grep("&",effects)
for(a in and_index){
  tmp_effects <- strsplit(effects[a], '&')[[1]]
  tmp_genes <- rep(genes[a], length(tmp_effects))
  effects <- c(effects, tmp_effects)
  genes <- c(genes, tmp_genes)
}
effects <- effects[-and_index]
genes <- genes[-and_index]
Effect_gene <- data.frame(EFFECT = effects,GENE = genes)

snpEff_table_Effect_table <- as.data.frame(table(Effect_gene$EFFECT)) 
colnames(snpEff_table_Effect_table) <- c('Mutation', 'Frequency')
write.table(snpEff_table_Effect_table, paste0(path, 'Final_Call-PASS_effects.txt'), sep = '\t', quote = F)
### Only differnetiated sites 
effects <- unlist(strsplit(snpEff_table_differentiated_sites$Effect, ':'))
genes <- unlist(strsplit(snpEff_table_differentiated_sites$Gene, ':'))
and_index <- grep("&",effects)
for(a in and_index){
  tmp_effects <- strsplit(effects[a], '&')[[1]]
  tmp_genes <- rep(genes[a], length(tmp_effects))
  effects <- c(effects, tmp_effects)
  genes <- c(genes, tmp_genes)
}
effects <- effects[-and_index]
genes <- genes[-and_index]
Effect_gene <- data.frame(EFFECT = effects,GENE = genes)

snpEff_table_differentiated_sites <- merge(snpEff_table, differentiated_sites, by = 'ID')
write.table(snpEff_table_differentiated_sites, paste0(path, 'Differentiated_sites_F9_H11_WT-snpEff.txt'), col.names = T, row.names = F, quote = F, sep = '\t')
#Merging differentiated sites with snp_eff table 
snpEff_table_differentiated_sites_Effect_table <- as.data.frame(table(snpEff_table_differentiated_sites$Effect)) 
colnames(snpEff_table_differentiated_sites_Effect_table) <- c('Mutation', 'Frequency')
print(xtable(snpEff_table_differentiated_sites_Effect_table, display = c("s","d","d"), digits = 0, caption = 'Functional annotation of sites with allele frequency differnece between F9 and H11/WT', lable = 'table:snpEff_differentiated_sites'), format.args = list(big.mark = " ", decimal.mark = "."))
##
##DSG2 mutations
snpEff_gene_mutation_list <- read.csv(paste0(path,'Final_Call-PASS.genes.txt'), sep = '\t', header =T)
DSG2_mutation <- snpEff_gene_mutation_list[(snpEff_gene_mutation_list$GeneName == 'DSG2'),]
kable(DSG2_mutation)
region <- extractRegion(data = Allele_freq, chr = coordinate[1], start = as.numeric(coordinate[2]), end = as.numeric(coordinate[3]), distance = 10000)
snpEff_table_AF <- merge(region, snpEff_table, by = 'ID')
DSG2_mutation_AF <- snpEff_table_AF[,c('Chr.x', 'Position', 'F9', 'H11', 'WT', 'Effect', 'Gene')]
write.table(DSG2_mutation_AF, paste0(path, 'DSG2_Mutation_AF.txt'), col.names = T, row.names = F, sep = '\t', quote = F)
print(xtable(DSG2_mutation_AF, display = c("d","d","d", "f","f","f", "s","s"), digits = 2 , caption = 'Identified mutations in DSG2 region and their functional impact.', label = 'table:DSG2_snpEff_AF'), format.args = list(big.mark = " ", decimal.mark = "."))
##
```



# OLD:  
## Comparing genotypes
To compare the differences between samples, we first extracted the genotypes of each sampels from high quality variants selected in previous step.  
```{bash}
#Extract genotypes 
module load bioinfo-tools vcftools
vcftools --vcf Final_Call-PASS_snpEff.vcf --012 --out Final_Call-PASS_snpEff
#Create bed file from genotypes
bash ~/private/Pipelines/transpose-matrix.sh Final_Call-PASS_snpEff.012
```

Then we generate the frequency of the genotypes  
```{r, eval = F, echo = T}
library(reshape2)
data <- read.table('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/snpEff/Final_Call-PASS_snpEff.012.t.bed', header = F)
indv <- read.table('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/snpEff/Final_Call-PASS_snpEff.012.indv', header = F)
colnames(data) <- c('Chr', 'Start', 'End', as.character(indv$V1))
Genotype_concat <- paste0(data$`SJ-2370-HAP1-DSG-F9`, data$`SJ-2370-HAP1-DSG-H11`, data$`SJ-2370-HAP1-WT`)
Genotype_concat_table <- (as.data.frame(table(Genotype_concat)))
Genotype_concat_table <- Genotype_concat_table[grep('-1',Genotype_concat_table$Genotype_concat, invert = T),]
rownames(Genotype_concat_table) <- Genotype_concat_table$Genotype_concat
pdf('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/snpEff/Final_Call-PASS_snpEff.012.pdf', width = 10, height = 5)
barplot(as.matrix(Genotype_concat_table$Freq) ~ as.matrix(Genotype_concat_table$Genotype_concat), ylim = c(0,50000), xlab = 'Genotype of F9;H11;WT respectively', ylab = 'Freq (ylim=c(0,50000))', main = 'Genotypes frequency')
dev.off()


#Genotype freq by chromosome
Genotype_concat_by_chr_melt <- melt(data[,c(1,7)])
Genotype_concat_by_chr_melt <- Genotype_concat_by_chr_melt[grep('-1', Genotype_concat_by_chr_melt$Genotype_concat,invert = T),]
Genotype_concat_by_chr_melt <- Genotype_concat_by_chr_melt[(as.numeric(Genotype_concat_by_chr_melt$Chr) <= 22),]
Genotype_concat_by_chr_melt_df <- tbl_df(Genotype_concat_by_chr_melt)
t <- Genotype_concat_by_chr_melt_df %>% group_by(Chr, Genotype_concat) %>% summarise(n = n()) %>% mutate(freq = n *100/ sum(n))
ggplot(t, aes(x=Chr, y=freq, fill = as.factor(Genotype_concat))) + geom_bar(position="stack", stat="identity") + labs(fill = "Genotype")

#DSG2 (chr18:29,078,027-29,128,814)
DSG2 <- tbl_df(data[(data$Chr == 18 &
                data$Start >=29078027 &
                data$End <= 29128814),])
DSG2 <- DSG2[grep('-1', DSG2$Genotype_concat,invert = T),]
DSG2_Genotype_concat <- DSG2 %>% group_by(Genotype_concat) %>% summarise(n = n()) %>% mutate( freq = n *100 / sum(n))
ggplot(DSG2_Genotype_concat, aes(x=Genotype_concat, y=freq)) + geom_bar(position="stack", stat="identity")
```

#DSG2 depth
Generating depth of coverage for samples over DSG2 gene:  
```{bash, eval = F}
cd /proj/snic2020-16-70/private/SMS_4947_20_CVA24v_receptor/intermediate/depth/
samtools depth -o F9_H11_WT_Q20_DSG2.depth -b DSG2.bed -Q 20 ../bam/SJ-2370-HAP1-DSG-F9.clean.dedup.bam ../bam/SJ-2370-HAP1-DSG-H11.clean.dedup.bam ../bam/SJ-2370-HAP1-WT.clean.dedup.bam
```

Visualising the depth  
```{r, eval = F}
library(zoo)
library(tidyverse)
DSG2_depth <- read.table('~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/depth/F9_H11_WT_Q20_DSG2_18_29.06_29.13.depth', header = F)
colnames(DSG2_depth) <- c('Chr','Pos','F9', 'H11', 'WT')
plot(DSG2_depth$Pos, DSG2_depth$F9, type = 'l', bty = 'n', xlab = 'Chr18', ylab = 'raw depth')
rect(xleft = 29078027, xright = 29128814, ybottom = -2, ytop = max(DSG2_depth[,3:6])+10), rgb(0,0,1,alpha=0.1))

Pos <- rep(1:nrow(DSG2_depth),1)
DSG2_depth_1kb <- cbind(Pos,DSG2_depth[,-c(1:2)])
DSG2_depth_1kb <- as.data.frame(rollapply(DSG2_depth_1kb,500,mean,by=500))
DSG2_depth_1kb$Pos <- seq(DSG2_depth$Pos[1], DSG2_depth$Pos[nrow(DSG2_depth)],500)[-1]

layout(matrix(c(1,2,3,4), 4, 1, byrow = TRUE), heights = c(30,30,30,70))
par(mar = c(3,5,0,5))
plot(DSG2_depth_1kb$Pos, DSG2_depth_1kb$F9, type = 'l', bty = 'n', xlab = 'Chr18', ylab = '1-kb raw depth', ylim = c(19,60))
lines(DSG2_depth_1kb$Pos, DSG2_depth_1kb$H11, col = 'green')
lines(DSG2_depth_1kb$Pos, DSG2_depth_1kb$WT, col = 'red')
statusCol <- gl(3, 1, labels=c('black', 'green', 'red'),length=3) %>% as.vector()
rect(xleft = 29078027, xright = 29128814, ybottom = 17, ytop = 70, col= rgb(1,0,0,alpha=0.1), border = NA)
par(xpd = T)
legend(par()$usr[2] + 0.2, par()$usr[4], legend=c('F9', 'H11', 'WT'), pch=20, col=statusCol, border = "white")

gff=read.table("~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/data/metadata/annotation/GSD2.gtf", header = F)
colnames(gff)=c("Chr","Source","Feature","Start","End","Score","Strand","Frame","attribute")

pos_s=55; #start position to plot gene (Y-axis)
pos_e=60 #end position to plot gene (Y-axis)
iter=round((pos_e-pos_s)/5,0) #define distance between consecutive genes (Y-axis)
pos_s_iter=pos_s
cntr=1

for(i in 1:nrow(gff))
{
  if(gff$Feature[i]=="gene")
  {
    print(pos_s_iter)
    print(cntr)
    x=gff$Attribute[i]
    gene_name=sub("ID.*Name=",x,replacement = "")
    if(gff$Strand[i]=="+")
    {
      arrows(x0=gff$Start[i],y0=pos_s_iter,x1=gff$End[i],y1=pos_s_iter,angle = 0.5,lwd=2)
      text(x=gff$Start[i]+(gff$End[i]-gff$Start[i])/2,labels = "",y=5+pos_s_iter,cex=0.7,font=4)
    }
    if(gff$Strand[i]=="-")
    {
      arrows(x1=gff$Start[i],y0=pos_s_iter,x0=gff$End[i],y1=pos_s_iter,angle = 0.5,lwd=2)
      text(x=gff$Start[i]+(gff$End[i]-gff$Start[i])/2,labels = gene_name,y=5+pos_s_iter,cex=0.7,font=4)
    }
    if(cntr<=5)
    {
      pos_s_iter=pos_s_iter+iter; 
      cntr=cntr+1
    }
    if(cntr==6)
    {
      pos_s_iter=pos_s
      cntr=1
    }

  }
  if(gff$Feature[i]=="exon"){
    segments(x0 = gff$Start[i], 
             x1 = gff$End[i],
             y0 = pos_s,
             y1 = pos_s, lwd = 8)
  }
}

sgRNA_blat_all <- data.frame(sgRNA = c("Exon2_sgRNA_F", "Exon2_sgRNA_R", "Exon3_sgRNA_F", "Exon3_sgRNA_R", "Exon5_sgRNA_F", "Exon8_sgRNA_F", "Exon8_sgRNA_F", "Exon8_sgRNA_F"), Start = c("29098200", "29098200", "29099828", "29099829", "29101067", "29104809", "20753162", "32596426"), End = c("29098221", "29098221", "29099850", "29099850", "29101087", "29104829", "20753177", "32596440"), Strand = c("+", "-", "+", "-", "+", "+", "-", "-"), Exon = c(2,2,3,3,5,5,8,8))
pos_s_iter = 55
cntr = 1
j = 1
for(r in c(2,3,5,8)){
  sgRNA_blat <- sgRNA_blat_all[(sgRNA_blat_all$Exon == r),]
  start = as.numeric(as.character(sgRNA_blat$Start[i]))
  end = as.numeric(as.character(sgRNA_blat$End[i]))
  for(i in 1:nrow(sgRNA_blat)){
    if(as.character(sgRNA_blat$Strand[i])=="+"){
      arrows(x0=start,y0=pos_s_iter+ cntr,x1=end,y1=pos_s_iter+ cntr,angle = 0.5,lwd=2)
      text(x=start+(end-start)/2,labels = as.character(sgRNA_blat$Exon[i]), y=pos_s_iter + cntr,cex=0.7,font=4)
    }
    if(sgRNA_blat$Strand[i]=="-"){
      arrows(x1=start,y0=pos_s_iter+ cntr,x0=end,y1=pos_s_iter + cntr,angle = 0.5,lwd=2, col = 'black')
      text(x=start+(end-start)/2-100,labels = as.character(sgRNA_blat$Exon[i]),y=pos_s_iter+ cntr,cex=0.7,font=4, co = 'black')
    }
  }
  if(j == 1){
    cntr = -2
    j = 2
  }else{
    cntr = 2
    j = 1
  }
}
#chr18:29,098,180-29,104,890


##ggbio
library(ggbio)
library(Homo.sapiens)
library(reshape2)

coordinate <- c(18, 29078027, 29128814)

path <- '~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/Variant_Calling/Genotyping/'
Allele_freq <- read.table(paste0(path,'Final_Call-PASS_snpEff_chr8_DSG2_50k_freq.bed'), head = F)
colnames(Allele_freq) <- c('Chr', 'Start', 'End', 'Info', 'Ref', 'Alt', 'QUAL', 'Filter', 'F9', 'H11', 'WT')
Allele_freq$ID <- paste0(Allele_freq$Chr, "_", Allele_freq$End)
region <- extractRegion(data = Allele_freq, chr = coordinate[1], start = as.numeric(coordinate[2]), end = as.numeric(coordinate[3]), distance = 10000)
region <- melt(Allele_freq[,c('End', 'F9', 'H11', 'WT')], id.vars = 'End', variable.name = 'Sample')
colnames(region) <- c('Position', 'Sample', 'Ref_Freq')

#load gene symbol : GRanges, one gene/row
data(genesymbol, package = "biovizBase")
#retrieve information of the gene of interest
wh <- genesymbol[c("DSG2")]
wh <- range(wh, ignore.strand = F)
#Plot the different transcripts  for our gene of interest
p.txdb <- autoplot(Homo.sapiens, which = wh)

#Depth heatmap
#Normalise the depth
genome_avg_depth <- data.frame(Sample = c('F9', 'H11', 'WT'), Avg_depth = c(35.97, 48.49, 34.75))
DSG2_depth_1kb_normalised <- DSG2_depth_1kb
DSG2_depth_1kb_normalised[,2:4] <- DSG2_depth_1kb[,2:4]/(genome_avg_depth$Avg_depth/max(genome_avg_depth$Avg_depth))
region <- melt(DSG2_depth_1kb_normalised[(DSG2_depth_1kb_normalised$Pos >= as.numeric(coordinate[2])),], id.vars = 'Pos')
colnames(region) <- c('Position', 'Sample', 'Depth')
depth_heatmap <- ggplot(data = region, aes ( x = Position, y = as.factor(Sample), fill = Depth, )) + geom_tile() + ylab(label = 'Sample') + xlab(label = 'chr18') + theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "white"), panel.background =  element_blank())

gr18 <- GRanges("chr18", IRanges(29078027, 29128814))
tks <- tracks(gene = p.txdb, Reference_Freq = p_freq, heights = c(1, 4)) + xlim(gr18) + theme_tracks_sunset()
tks
```


#SV analyses
Insert size standard deviation and mean  
```{r, eval = F}
library(knitr)
SD_Mean <- data.frame(Sample = c('F9', 'H11', 'WT'), 
                      Mean = c(377,375,407),
                      SD = c(205,204,191))
kable(SD_Mean)
```
Parsing bam file for SV (read orientation information)  
```{bash, eval = F}
~/private/PERL/bam-parsing/bam-parsing-Inversion-V6-print-clusters-by-count.pl SJ-2370-HAP1-DSG-H11.clean.dedup.bam ~/SMS_4947_20_CVA24v_receptor/code/../data/meta_data/reference/human_g1k_v37.fasta.fai 204 375 18 >SJ-2370-HAP1-DSG-H11_18_29.06_29.13_Inversion.txt &

~/private/PERL/bam-parsing/bam-parsing-Inversion-V6-print-clusters-by-count.pl SJ-2370-HAP1-DSG-F9.clean.dedup.bam ~/SMS_4947_20_CVA24v_receptor/code/../data/meta_data/reference/human_g1k_v37.fasta.fai 250  377 18 >SJ-2370-HAP1-DSG-F9_18_29.06_29.13_Inversion.txt &

~/private/PERL/bam-parsing/bam-parsing-Inversion-V6-print-clusters-by-count.pl SJ-2370-HAP1-WT.clean.dedup.bam ~/SMS_4947_20_CVA24v_receptor/code/../data/meta_data/reference/human_g1k_v37.fasta.fai 204 375 18 >SJ-2370-HAP1-WT_18_29.06_29.13_Inversion.txt &
```

Visualising the disoriented reads  
```{r, eval = F}
path <- '~/Dropbox/NBIS/Projects/SMS_4947_20_CVA24v_receptor/intermediate/bam/'
samples <- c('F9', 'H11', 'WT')
for(s in samples){
  print(s)
  par(mar = c(0,5,3,0))
  plot(x = c(min(DSG2_depth_1kb$Pos), max(DSG2_depth_1kb$Pos)), y = c(0,30), bty = 'n', xaxt = 'n', xlab = '', ylab = 'insert size / 100', main = paste0(s, ' (Reads supporting inversion)'), type = 'n')
  tmp <- read.table(paste0(path, 'SJ-2370-HAP1-DSG-', s,'_18_29.06_29.13_Inversion.txt'), header = T)
  tmp <- tmp[!is.na(tmp$X3locdev.reads),]
  tmp <- tmp[(tmp$Disoriented_reads>1),]
  for (r in 1:nrow(tmp)){
    starts <- as.integer(unlist(strsplit(x = as.character(tmp[r,'X5locdev.reads']), split = ',')))
    ends <- as.integer(unlist(strsplit(x = as.character(tmp[r,'X3locdev.reads']), split = ',')))
    for(i  in  1:length(starts)){
  		m <- round(median(as.integer(starts[i]),as.integer(ends[i])))
  		distance <- abs(as.integer(starts[i]) - as.integer(ends[i]))/100
  		segments(x0 = starts[i],y0 = 0,x1 = m,y1 = distance,lwd=1)
  		segments(x1 = ends[i],y1 = 0,x0 = m,y0 = distance,lwd=1)
    }
  }
}

plot(x = c(min(DSG2_depth_1kb$Pos), max(DSG2_depth_1kb$Pos)), y = c(0,30), bty = 'n',  xlab = '', ylab = 'insert size / 100', main = paste0(s, ' (Reads supporting inversion)'), type = 'n')
```